### DISCUSSION TITLE
阿里云盘open填写客户端id和秘钥提示 failed to refresh token: invalid client_id

### DISCUSSION THREAD

**Discussion by 520yxl** [AlistGo/alist#4578] [Q&A] [answered]
阿里云盘open填写客户端id和秘钥提示 failed to refresh token: invalid client_id

**Comment by Masterlovsky** [comment]
检查一下你的token是不是过期了，重新获取一下

**Reply by 520yxl** [reply]
> 检查一下你的token是不是过期了，重新获取一下

重新获取了,只要填写了自己申请的客户端ID和密钥就不行

**Comment by anwen-anyi** [comment]
无效客户端ID 你自己申请的客户端ID和秘钥吗

**Reply by 520yxl** [reply]
> 无效客户端ID 你自己申请的客户端ID和秘钥吗

是我自己申请的

**Reply by anwen-anyi** [reply]
截图看看你的填写方式

**Comment by xhofe** [comment, marked as answer]
使用自己的client 需要使用自己的client获取 refresh_token， 方法参加阿里云盘官方文档：https://www.yuque.com/aliyundrive/zpfszx/efabcs
alist文档中的工具暂只支持使用alist的client。

**Reply by zhychen1173** [reply]
> 使用自己的client 需要使用自己的client获取 refresh_token， 方法参加阿里云盘官方文档：https://www.yuque.com/aliyundrive/zpfszx/efabcs alist文档中的工具暂只支持使用alist的client。

我用 https://alist.nn.ci/tool/aliyundrive/request.html 使用自己的client_id 和 client_secret ，然后Scan QRcode ，手机app 授权之后，点击 I have scan，跳转页面报错，报：InvalidClientId: invalid client_id
InvalidClientId: invalid client_id

阿里云盘手机APP授权完成后，进到APP的授权管理页面，看到是已经授权成功的。就是https://alist.nn.ci/tool/aliyundrive/request.html  跳转获取token的页面报 InvalidClientId: invalid client_id 的错误。

请教应该如何解决？谢谢。

**Reply by 520yxl** [reply]
检查client_id是否正确

**Reply by zhychen1173** [reply]
> 检查client_id是否正确

id正确的，阿里云盘app里可以看到已经授权成功。

**Reply by 520yxl** [reply]
> > 检查client_id是否正确
> 
> id正确的，阿里云盘app里可以看到已经授权成功。

正确的话，正常的来说就不会再报无效client_id

**Reply by 520yxl** [reply]
你换成Go to login试一下

**Comment by 520yxl** [comment]
已根据阿里云盘官方文档解决了

**Reply by shuheng-2834** [reply]
如何解决的呢，我也遇到了一模一样的问题
